### PR TITLE
Add enable-state-field-trie for e2e

### PR DESCRIPTION
**Pull request by prestonvanloon**
We're experimenting with this flag in production. It should be run in e2e for some time before being released to all. 

- [x] Do not recompute trie if no new indices have been added.
- [x] Add helper to remove duplicate indices for slices.

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Fails e2e consistently. @nisdas do you have any idea?

File: shared/sliceutil/slice.go [hunk 1]
```diff
@@ -83,6 +83,22 @@ func UnionUint64(s ...[]uint64) []uint64 {
 	return set
 }
 
+// CleanUint64 removes duplicate indices from
+// the slice.
+func CleanUint64(a []uint64) []uint64 {
```
Comment (terencechain) [inline shared/sliceutil/slice.go]: Clean as a term is ambiguous in this case. How about `SetUint64`?

File: shared/sliceutil/slice.go [hunk 2]
```diff
@@ -83,6 +83,22 @@ func UnionUint64(s ...[]uint64) []uint64 {
...
+	// remove duplicate indexes
```
Comment (terencechain) [inline shared/sliceutil/slice.go]:
```suggestion
	// Remove duplicates indices.
```

File: shared/sliceutil/slice.go (hunk 1 above)
Comment (nisdas) [reply shared/sliceutil/slice.go]: `SetUint64` also doesn't describe what this method actually does, its too generic. Any other names you think would work ? I couldn't find a good name to describe it

Comment (terencechain) [reply shared/sliceutil/slice.go]: Set means deduplicates (ie pyhton, java)  in most context.  What meaning of Set are you referring to that doesn't describe?

Comment (nisdas) [reply shared/sliceutil/slice.go]: `Set` meaning setters/getters . `SetUint64` reads as though your are setting the value to these indices, which wouldn't be wholly correct.  Reading your link you are referring to a mathematical set, which would make more sense.

Comment (nisdas) [reply shared/sliceutil/slice.go]: ok changed, done

Comment (terencechain) [reply shared/sliceutil/slice.go]: I see. I was more referring to spec uses Set for duplicates: https://github.com/ethereum/eth2.0-specs/blob/dev/specs/phase0/beacon-chain.md#get_attesting_indices

But I do see the confusion :)

**Comment by prestonvanloon** [comment]
Thanks for the fixes @nisdas

**Pull request merged**